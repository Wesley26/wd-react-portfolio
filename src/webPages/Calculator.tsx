import { FC, useEffect, useState } from 'react';
import PageContents from './contentMaster/PageContents';
import MainPageAnimations from './contentMaster/MainPageAnimations';
import { motion } from 'framer-motion';

import CalculatorBase from './calculatorComponents/CalculatorBase';

const Calculator:FC = () => {

    const pageContents = PageContents();
    const mainPageAnimations = MainPageAnimations();

    const [visible, setVisible] = useState<boolean>(false);

    useEffect(() => {

        setVisible(false);
        setVisible(true);

    }, []);

    return (

        <div className="font-body bg-body_lightGray m-6 shadow-xl">

            <> { visible ? (
                <motion.div
                    initial={mainPageAnimations.initialStyle}
                    animate={mainPageAnimations.animateStyle}
                    transition={mainPageAnimations.transitionStyle}
                    className="px-5 md:px-40 py-3"
                >

                    <CalculatorBase 
                        addSign={pageContents.calculatorContent.addSign}
                        deleteBTN={pageContents.calculatorContent.deleteBTN}
                        divideSign={pageContents.calculatorContent.divideSign}
                        equalsSign={pageContents.calculatorContent.equalsSign}
                        multiplySign={pageContents.calculatorContent.multiplySign}
                        periodSign={pageContents.calculatorContent.periodSign}
                        subtractSign={pageContents.calculatorContent.subtractSign}
                        zeroSign={pageContents.calculatorContent.zeroSign}
                    />

                </motion.div>
            ) : null }
            </>

            <div className="py-6" />

            <>
                { visible === true ? (
                <div 
                    className="text-xl p-3 mx-3 md:mx-12 lg:mx-48 my-12 border-2 border-black flex rounded-full justify-center hover:bg-gray-300 active:bg-gray-500"
                    onClick={() => { setVisible(false); }}
                >
                    {pageContents.calculatorContent.closeCal}
                </div> ) : (
                <div 
                    className="text-xl p-3 mx-3 md:mx-12 lg:mx-48 my-12 border-2 border-black flex rounded-full justify-center hover:bg-gray-300 active:bg-gray-500"
                    onClick={() => { setVisible(true); }}
                >
                    {pageContents.calculatorContent.openCal}
                </div> )}
            </>

            <div className="py-6" />

        </div>

    );

};

export default Calculator;