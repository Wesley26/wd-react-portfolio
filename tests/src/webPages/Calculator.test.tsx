import { render } from '@testing-library/react';
import Calculator from '../../../src/webPages/Calculator';
import '../../src/index.css';

it(`render ContactMe.tsx test component`, () => {

    const view = (render(<Calculator />));

    expect(view).toMatchSnapshot();

});