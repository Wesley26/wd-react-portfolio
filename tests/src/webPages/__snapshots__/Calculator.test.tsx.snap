// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render ContactMe.tsx test component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="font-body bg-body_lightGray m-6 shadow-xl"
      >
         
        <div
          class="px-5 md:px-40 py-3"
          style="opacity: 0; transform: translateX(0px) translateY(-20px) translateZ(0);"
        >
          <div
            class="flex flex-col px-5 py-3 min-h-screen justify-center items-center font-calculator bg-cal_light border-black border-2 rounded-md shadow-lg"
          >
            <div
              class="w-full max-w-400pix bg-cal_light_altOne rounded-16pix overflow-hidden shadow-md"
            >
              <div
                class="p-16pix text-right bg-cal_tirshiary text-cal_light text-24e font-light"
              >
                  
                0
              </div>
              <div
                class="flex font-calculator text-cal_light"
              >
                <div
                  class="flex-1 p-4 border-none text-20e bg-cal_primary font-bold hover:bg-cal_hover active:bg-cal_active"
                >
                  DEL
                </div>
              </div>
              <div
                class="flex font-calculator text-cal_light"
              >
                <div
                  class="flex-1 p-4 border-none text-20e bg-cal_primary font-bold hover:bg-cal_hover active:bg-cal_active"
                >
                  /
                </div>
                <div
                  class="flex-1 p-4 border-none text-20e bg-cal_primary font-bold hover:bg-cal_hover active:bg-cal_active"
                >
                  *
                </div>
                <div
                  class="flex-1 p-4 border-none text-20e bg-cal_primary font-bold hover:bg-cal_hover active:bg-cal_active"
                >
                  +
                </div>
                <div
                  class="flex-1 p-4 border-none text-20e bg-cal_primary font-bold hover:bg-cal_hover active:bg-cal_active"
                >
                  -
                </div>
              </div>
              <div
                class="flex flex-wrap text-cal_light"
              >
                <div
                  class="flex-1And33percent p-4 border-none text-20e max-w-33percent bg-cal_dark hover:bg-cal_hover active:bg-cal_active"
                >
                  1
                </div>
                <div
                  class="flex-1And33percent p-4 border-none text-20e max-w-33percent bg-cal_dark hover:bg-cal_hover active:bg-cal_active"
                >
                  2
                </div>
                <div
                  class="flex-1And33percent p-4 border-none text-20e max-w-33percent bg-cal_dark hover:bg-cal_hover active:bg-cal_active"
                >
                  3
                </div>
                <div
                  class="flex-1And33percent p-4 border-none text-20e max-w-33percent bg-cal_dark hover:bg-cal_hover active:bg-cal_active"
                >
                  4
                </div>
                <div
                  class="flex-1And33percent p-4 border-none text-20e max-w-33percent bg-cal_dark hover:bg-cal_hover active:bg-cal_active"
                >
                  5
                </div>
                <div
                  class="flex-1And33percent p-4 border-none text-20e max-w-33percent bg-cal_dark hover:bg-cal_hover active:bg-cal_active"
                >
                  6
                </div>
                <div
                  class="flex-1And33percent p-4 border-none text-20e max-w-33percent bg-cal_dark hover:bg-cal_hover active:bg-cal_active"
                >
                  7
                </div>
                <div
                  class="flex-1And33percent p-4 border-none text-20e max-w-33percent bg-cal_dark hover:bg-cal_hover active:bg-cal_active"
                >
                  8
                </div>
                <div
                  class="flex-1And33percent p-4 border-none text-20e max-w-33percent bg-cal_dark hover:bg-cal_hover active:bg-cal_active"
                >
                  9
                </div>
                <div
                  class="flex-1And33percent p-4 border-none text-20e max-w-33percent bg-cal_dark hover:bg-cal_hover active:bg-cal_active"
                >
                  0
                </div>
                <div
                  class="flex-1And33percent p-4 border-none text-20e max-w-33percent bg-cal_dark hover:bg-cal_hover active:bg-cal_active"
                >
                  .
                </div>
                <div
                  class="flex-1And33percent p-4 border-none text-20e max-w-33percent bg-cal_dark hover:bg-cal_hover active:bg-cal_active"
                >
                  =
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="py-6"
        />
        <div
          class="text-xl p-3 mx-3 md:mx-12 lg:mx-48 my-12 border-2 border-black flex rounded-full justify-center hover:bg-gray-300 active:bg-gray-500"
        >
          Close Calculator
        </div>
        <div
          class="py-6"
        />
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="font-body bg-body_lightGray m-6 shadow-xl"
    >
       
      <div
        class="px-5 md:px-40 py-3"
        style="opacity: 0; transform: translateX(0px) translateY(-20px) translateZ(0);"
      >
        <div
          class="flex flex-col px-5 py-3 min-h-screen justify-center items-center font-calculator bg-cal_light border-black border-2 rounded-md shadow-lg"
        >
          <div
            class="w-full max-w-400pix bg-cal_light_altOne rounded-16pix overflow-hidden shadow-md"
          >
            <div
              class="p-16pix text-right bg-cal_tirshiary text-cal_light text-24e font-light"
            >
                
              0
            </div>
            <div
              class="flex font-calculator text-cal_light"
            >
              <div
                class="flex-1 p-4 border-none text-20e bg-cal_primary font-bold hover:bg-cal_hover active:bg-cal_active"
              >
                DEL
              </div>
            </div>
            <div
              class="flex font-calculator text-cal_light"
            >
              <div
                class="flex-1 p-4 border-none text-20e bg-cal_primary font-bold hover:bg-cal_hover active:bg-cal_active"
              >
                /
              </div>
              <div
                class="flex-1 p-4 border-none text-20e bg-cal_primary font-bold hover:bg-cal_hover active:bg-cal_active"
              >
                *
              </div>
              <div
                class="flex-1 p-4 border-none text-20e bg-cal_primary font-bold hover:bg-cal_hover active:bg-cal_active"
              >
                +
              </div>
              <div
                class="flex-1 p-4 border-none text-20e bg-cal_primary font-bold hover:bg-cal_hover active:bg-cal_active"
              >
                -
              </div>
            </div>
            <div
              class="flex flex-wrap text-cal_light"
            >
              <div
                class="flex-1And33percent p-4 border-none text-20e max-w-33percent bg-cal_dark hover:bg-cal_hover active:bg-cal_active"
              >
                1
              </div>
              <div
                class="flex-1And33percent p-4 border-none text-20e max-w-33percent bg-cal_dark hover:bg-cal_hover active:bg-cal_active"
              >
                2
              </div>
              <div
                class="flex-1And33percent p-4 border-none text-20e max-w-33percent bg-cal_dark hover:bg-cal_hover active:bg-cal_active"
              >
                3
              </div>
              <div
                class="flex-1And33percent p-4 border-none text-20e max-w-33percent bg-cal_dark hover:bg-cal_hover active:bg-cal_active"
              >
                4
              </div>
              <div
                class="flex-1And33percent p-4 border-none text-20e max-w-33percent bg-cal_dark hover:bg-cal_hover active:bg-cal_active"
              >
                5
              </div>
              <div
                class="flex-1And33percent p-4 border-none text-20e max-w-33percent bg-cal_dark hover:bg-cal_hover active:bg-cal_active"
              >
                6
              </div>
              <div
                class="flex-1And33percent p-4 border-none text-20e max-w-33percent bg-cal_dark hover:bg-cal_hover active:bg-cal_active"
              >
                7
              </div>
              <div
                class="flex-1And33percent p-4 border-none text-20e max-w-33percent bg-cal_dark hover:bg-cal_hover active:bg-cal_active"
              >
                8
              </div>
              <div
                class="flex-1And33percent p-4 border-none text-20e max-w-33percent bg-cal_dark hover:bg-cal_hover active:bg-cal_active"
              >
                9
              </div>
              <div
                class="flex-1And33percent p-4 border-none text-20e max-w-33percent bg-cal_dark hover:bg-cal_hover active:bg-cal_active"
              >
                0
              </div>
              <div
                class="flex-1And33percent p-4 border-none text-20e max-w-33percent bg-cal_dark hover:bg-cal_hover active:bg-cal_active"
              >
                .
              </div>
              <div
                class="flex-1And33percent p-4 border-none text-20e max-w-33percent bg-cal_dark hover:bg-cal_hover active:bg-cal_active"
              >
                =
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="py-6"
      />
      <div
        class="text-xl p-3 mx-3 md:mx-12 lg:mx-48 my-12 border-2 border-black flex rounded-full justify-center hover:bg-gray-300 active:bg-gray-500"
      >
        Close Calculator
      </div>
      <div
        class="py-6"
      />
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
